compiler/modules/importer.nim

```diff
@@ -377,7 +377,29 @@ proc evalImport*(c: PContext, n: PNode): PNode =
   var hasError = false
 
   for it in n.sons:
-    if it.kind == nkInfix and it.len == 3 and it[2].kind == nkBracket:
+    if it.kind == nkPrefix and it.len == 2 and it[1].kind == nkBracket:
+      let
+        sep = it[0]
+        impTmpl =
+          block:
+            let t = newNodeI(nkPrefix, it.info, 2)
+            t[0] = sep
+            t
+          ## `impTmpl` is copied/instanced in a loop below including setting
+          ## the second child
+
+      for x in it[1]:
+        let imp = copyTree(impTmpl)
+
+        # transform `./[a as b] to `./a as b`
+        imp[1] =
+          if x.kind == nkInfix and x[0].ident.s == "as":
+            x[1]
+          else:
+            x
```

Comment:
```suggestion
      for x in it[1].items:
        let imp = newTreeI(nkPrefix, it.info,
          it[0],
          # transform `./[a as b] to `./a as b`
          if x.kind == nkInfix and x[0].ident.s == "as":
            x[1]
          else:
            x
        )
```

I know this code was copied from below, but the suggestion should improve clarity a bit. Creating an incomplete temporary tree just to repeatedly copy it afterwards is confusing (besides being inefficient).

Aside: I don't think the `transform x as y` comment is correct. If I'm reading the code correctly, what happens is that `./[x as a]` is turned into `./x` (the `nkInfix` handling has the same problem).